src/Microsoft.DotNet.Build.Tasks/GenerateTestOutputDirList.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateTestOutputDirList : Task
+    {
+        [Required]
+        public string OutputDirectory { get; set; }
+
+        [Required]
+        public ITaskItem[] TestOutputItems { get; set; }
+
+        [Required]
+        public string AssemblyName { get; set; }
+
+        public override bool Execute()
+        {
+
+            if (TestOutputItems.Length == 0)
+            {
+                throw new InvalidOperationException("TestOutputItems property is empty.");
+            }
+
+            if (!Directory.Exists(OutputDirectory))
+            {
+                Directory.CreateDirectory(OutputDirectory);
+            }
+
+            string testOutputDirList = Path.Combine(OutputDirectory, AssemblyName + ".txt");
+
+            string selectedDllPath = string.Empty;
+            foreach (var item in TestOutputItems)
+            {
+                selectedDllPath = item.ItemSpec.Replace('\\', '/');
+            }
+
+            HashSet<string> uniqueTestOutputDirectories = new HashSet<string>();
+            if (File.Exists(testOutputDirList))
```

Comment:
Does this both read and write the same file? I'm pretty sure it does, if I'm reading the code correctly, and tasks shouldn't do that, since it makes incremental builds impossible, and it's difficult to debug what's happening.